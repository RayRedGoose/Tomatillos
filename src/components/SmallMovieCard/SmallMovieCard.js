import React, { Component } from 'react';
import './SmallMovieCard.scss'
import { Redirect } from 'react-router-dom';
import deleteIcon from '../../assets/images/delete.svg';

class SmallMovieCard extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isRedirected: false
    }
  }

  redirect = () => {
    this.setState({isRedirected: true})
  }

  render() {
    const { id, img, title, rate, deleteButton } = this.props

    const headerStyles = {
      backgroundImage: `url("${img}")`,
      backgroundSize: '100%',
      backgroundPosition: 'center',
      width: '230px',
      height: '300px'
    }

    return (
      (this.state.isRedirected)
        ? <Redirect to={`/movies/${id}`} />
        : <section className="small-movie-card">
          <header style={headerStyles}>
            {deleteButton && <img src={deleteIcon} alt="delete icon" />}
          </header>
          <p className="rate-number">{ rate }</p>
          <div className="title-block">
            <p className="title" onClick={this.redirect}>{ title }</p>
          </div>
        </section>
    );
  }
}

export default SmallMovieCard;
