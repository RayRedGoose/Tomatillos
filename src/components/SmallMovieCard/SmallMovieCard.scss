@import "../../variables.scss";

.small-movie-card {
  box-shadow: 0 0 24px 0 rgba(0, 0, 0, 0.5);
  cursor: pointer;
  margin: 0 20px 40px;
  p {
    color: #fff;
    font-size: 18px;
    font-weight: 700;
    text-align: center;
    &.rate-number {
      background-color: $main-accent-color;
      border-radius: 50%;
      margin: -10px auto;
      padding: 0.4em 3.5em;
      width: fit-content;
      position: relative;
      z-index: 3;
    }
  }
  .title-block {
    background: rgba(49, 59, 58, 0.8);
    backdrop-filter: blur(4px);
    display: flex;
    align-items: center;
    justify-content: center;
    margin: -10px auto 0;
    padding: 15px 10px;
    width: 210px; height: 70px;
    &>p {
      color: $titles-color;
      font-family: $large-text-font;
      font-size: 24px;
    }
  }
}
