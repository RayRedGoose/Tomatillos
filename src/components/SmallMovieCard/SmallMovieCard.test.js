import React from 'react';
import { shallow } from 'enzyme';
import SmallMovieCard from './SmallMovieCard';

describe("SmallMovieCard", () => {
  let card, mockProps, instance;

  beforeEach(() => {
    mockProps = {
      id: 1,
      img: 'https://some.url/img.jpg',
      title: 'Jumanji: The Next Level',
      rate: 4
    }

    card = shallow(
      <SmallMovieCard { ...mockProps } />
    )

    instance = card.instance()
  });

  it("should match snapshot with all data passed in correctly", () => {
    const cardWithDeleteButton = shallow(
      <SmallMovieCard { ...mockProps } deleteButton={true} />
    )
    expect(cardWithDeleteButton).toMatchSnapshot();
  });

  it("should match snapshot with all data passed in correctly except delete button", () => {
    expect(card).toMatchSnapshot();
  });

  it("should match snapshot with redirect router if redirect state is true", () => {
    card.setState({isRedirected: true});
    expect(card).toMatchSnapshot();
  });

  it("should call redirect after title was clicked", () => {
    const spy = jest.spyOn(instance, 'redirect').mockImplementation(() => {});
    instance.forceUpdate();

    card.find('.title').simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it("should change isRedirected state to true when redirect is called", () => {
    expect(card.state('isRedirected')).toEqual(false);

    instance.redirect();

    expect(card.state('isRedirected')).toEqual(true);
  });
});
