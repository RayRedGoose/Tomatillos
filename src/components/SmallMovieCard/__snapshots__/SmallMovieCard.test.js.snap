// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard should match snapshot with all data passed in correctly 1`] = `
<section
  className="small-movie-card"
>
  <header
    style={
      Object {
        "backgroundImage": "url(\\"https://some.url/img.jpg\\")",
        "backgroundPosition": "center",
        "backgroundSize": "100%",
        "height": "300px",
        "width": "230px",
      }
    }
  >
    <img
      alt="delete icon"
      src="delete.svg"
    />
  </header>
  <p
    className="rate-number"
  >
    4
  </p>
  <div
    className="title-block"
  >
    <p
      className="title"
      onClick={[Function]}
    >
      Jumanji: The Next Level
    </p>
  </div>
</section>
`;

exports[`SmallMovieCard should match snapshot with all data passed in correctly except delete button 1`] = `
<section
  className="small-movie-card"
>
  <header
    style={
      Object {
        "backgroundImage": "url(\\"https://some.url/img.jpg\\")",
        "backgroundPosition": "center",
        "backgroundSize": "100%",
        "height": "300px",
        "width": "230px",
      }
    }
  />
  <p
    className="rate-number"
  >
    4
  </p>
  <div
    className="title-block"
  >
    <p
      className="title"
      onClick={[Function]}
    >
      Jumanji: The Next Level
    </p>
  </div>
</section>
`;

exports[`SmallMovieCard should match snapshot with redirect router if redirect state is true 1`] = `
<Redirect
  to="/movies/1"
/>
`;
